src/Init/System/IO.lean

```diff
@@ -31,37 +31,61 @@ def IO.RealWorld : Type := Unit
 -/
 def EIO (ε : Type) : Type → Type := EStateM ε IO.RealWorld
 
-@[inline] def EIO.catchExceptions (x : EIO ε α) (h : ε → EIO Empty α) : EIO Empty α :=
-  fun s => match x s with
-  | EStateM.Result.ok a s     => EStateM.Result.ok a s
-  | EStateM.Result.error ex s => h ex s
-
 instance : Monad (EIO ε) := inferInstanceAs (Monad (EStateM ε IO.RealWorld))
 instance : MonadFinally (EIO ε) := inferInstanceAs (MonadFinally (EStateM ε IO.RealWorld))
 instance : MonadExceptOf ε (EIO ε) := inferInstanceAs (MonadExceptOf ε (EStateM ε IO.RealWorld))
 instance : OrElse (EIO ε α) := ⟨MonadExcept.orElse⟩
 instance [Inhabited ε] : Inhabited (EIO ε α) := inferInstanceAs (Inhabited (EStateM ε IO.RealWorld α))
 
+/-- An `EIO` monad that cannot throw exceptions. -/
+abbrev RealM := EIO Empty
+
+@[inline] def RealM.toEIO (self : RealM α) : EIO ε α :=
+  fun s => match self s with
+  | EStateM.Result.ok a s => EStateM.Result.ok a s
+
+instance : MonadLift RealM (EIO ε) := ⟨RealM.toEIO⟩
+
+@[inline] def EIO.toRealM (self : EIO ε α) : RealM (Except ε α) :=
+  fun s => match self s with
+  | EStateM.Result.ok a s     => EStateM.Result.ok (Except.ok a) s
+  | EStateM.Result.error ex s => EStateM.Result.ok (Except.error ex) s
+
+@[inline] def EIO.toRealM_ (self : EIO ε α) : RealM Unit :=
+  fun s => match self s with
+  | EStateM.Result.ok a s     => EStateM.Result.ok () s
+  | EStateM.Result.error ex s => EStateM.Result.ok () s
+
+@[inline] def EIO.catchExceptions (self : EIO ε α) (h : ε → RealM α) : RealM α :=
+  fun s => match self s with
+  | EStateM.Result.ok a s     => EStateM.Result.ok a s
+  | EStateM.Result.error ex s => h ex s
+
 open IO (Error) in
 abbrev IO : Type → Type := EIO Error
 
-@[inline] def EIO.toIO (f : ε → IO.Error) (x : EIO ε α) : IO α :=
-  x.adaptExcept f
+@[inline] def RealM.toIO (self : RealM α) : IO α :=
+  self
+
+@[inline] def EIO.toIO (f : ε → IO.Error) (self : EIO ε α) : IO α :=
+  self.adaptExcept f
 
-@[inline] def EIO.toIO' (x : EIO ε α) : IO (Except ε α) :=
-  EIO.toIO (fun _ => unreachable!) (observing x)
+@[inline] def EIO.toIO' (self : EIO ε α) : IO (Except ε α) :=
+  self.toRealM
 
-@[inline] def IO.toEIO (f : IO.Error → ε) (x : IO α) : EIO ε α :=
-  x.adaptExcept f
+@[inline] def IO.toEIO (f : IO.Error → ε) (self : IO α) : EIO ε α :=
+  self.adaptExcept f
```

Comment:
Why this unnecessary churn? The commit guidelines say to adapt to the style of the surrounding code, not the reverse. If we want to change the style of the stdlib, we should have a discussion first.

Reply:
@Kha the arguments names are not used anywhere in the Lean core and there doesn't appear to be, as far as I could tell, a consistent naming convention (e.g., `x` was used here, `fn` is used below in `unsafe*`,  `act` was used in `asTask`).  As such, I was under the impression the names weren't significant. Thus, I just went with my usual style (`self` for arguments intended to be used with dot notation) and adjusted close similar functions to match in order to introduce a little more consistency across them. I am happy to revert the name changes if desired, but then I would love some guidelines as to what convention I should be following for argument names for functions.

Reply:
I used `x` for any monadic value, but I'm happy with Leo's `act`. `fn` is strange indeed.

Reply:
@Kha so how would you like me to handle this? Would you like me to revert these changes and use `x`  (or `act`) instead in the new functions? Or is it fine enough as is?

Reply:
@Kha 

> I'd suggest to change every `self` to `act` just to not introduce a fourth style. You're welcome to open a thread in #lean4-dev about naming guidelines going forward.

done